src/refiners/foundationals/fuyu/tokenizer.py

```diff
@@ -0,0 +1,187 @@
+import gzip
+import json
+import re
+from pathlib import Path
+from typing import List, Tuple
+
+from numpy import round
+from torch import Tensor, tensor
+
+import refiners.fluxion.layers as fl
+
+
+class FuyuTokenizer(fl.Module):
+    """
+    Implement a Unigram Tokenizer based on a vocabulary file for Fuyu
+    """
+
+    def __init__(
+        self,
+        vocabulary_path: str | Path = Path(__file__).resolve().parent / "tokenizer.json.gz",
```

Comment:
Not reviewing everything, but this is a no-go, we will not add a 3 MB `json.gz` file to the source tree. This should be an asset that we deal with like the weights (downloaded externally).

Reply:
Understood, no problem i will change it this evening. Wanted to use the same logic as the CLIP tokenizer and the bpe_simple_vocab_16e6.txt.gz (even if i agree my vocab file is 2MB heavier)

Reply:
Yes, I know we do the same thing for CLIP. We use CLIP a lot so it simplified things initially. I am not sure we'd do the same thing today.

Reply:
I now create a .cache/refiners/fuyu-8b folder in the /home folder, save the tokenizer.json.gz there and retrieve it in the FuyuTokenizer class